x-pack/plugins/maps/public/classes/styles/vector/vector_style.tsx

```diff
@@ -186,10 +193,12 @@ export class VectorStyle implements IVectorStyle {
     descriptor: VectorStyleDescriptor | null,
     source: IVectorSource,
     layer: IVectorLayer,
+    customIcons?: CustomIcon[],
     chartsPaletteServiceGetColor?: (value: string) => string | null
   ) {
     this._source = source;
     this._layer = layer;
+    this._customIcons = customIcons ? customIconsToObject(customIcons) : undefined;
```

Comment:
Change to `this._customIcons = Object.fromEntries(customIcons.map((icon) => [icon.symbolId, icon]));`

No need to check customIcons is not provided, and inline `customIconsToObject` since its only used once and has a simple implemenation